Fix TMDB import failing when no IMDB ID is set for a movie

benedikt257 opened the pull request:
**Changes**
Makes use of the new TrySetProvider method to safeguard against cases in which the TMDB API returns null for the IMDB ID. 
This should ensure that the rest of the metadata setting proceeds as desired. 

**Issues**
Fixes #12884

benedikt257 commented:
I could not find any existing tests for the TmdbMovieProvider, is that right? If that is desired, I´d be happy to give it a go to add a few - but I haven´t worked with xUnit before, so it might take a while...

Gauvino commented:
This needs to be merged on 10.10 not master right ?

benedikt257 commented:
Thanks for the quick review @Gauvino!

> This needs to be merged on 10.10 not master right ?

True, what is the workflow for that? Should I just raise a PR to the `release-10.10.z` branch **in addition** to this one?
The documentation I see here: https://jellyfin.org/docs/general/contributing/release-procedure rather indicates that this PR should be tagged `stable-backport`?

Apologies if this is documented somewhere, as you can see this is my first PR here.

Gauvino commented:
> Thanks for the quick review @Gauvino!
> 
> 
> 
> > This needs to be merged on 10.10 not master right ?
> 
> 
> 
> True, what is the workflow for that? Should I just raise a PR to the `release-10.10.z` branch **in addition** to this one?
> 
> The documentation I see here: https://jellyfin.org/docs/general/contributing/release-procedure rather indicates that this PR should be tagged `stable-backport`?
> 
> 
> 
> Apologies if this is documented somewhere, as you can see this is my first PR here. 

Yes, just rebase you branch PR on 10.10 and it will automatically be backported to master

benedikt257 commented:
Thanks @Gauvino, I have rebased the branch onto release-10.10.z and changed the target of this PR accordingly. 
Hope this is alright now, thanks for your help.

Bond-009 commented on CONTRIBUTORS.md:
```diff
@@ -265,3 +265,4 @@
  - [0x25CBFC4F](https://github.com/0x25CBFC4F)
  - [Robert Lützner](https://github.com/rluetzner)
  - [Nathan McCrina](https://github.com/nfmccrina)
+ - [benedikt257](https://github.com/benedikt257)
```
This is on the wrong place, add it to the list of Jellyfin contributors instead of Emby

benedikt257 replied:
Sorry, didn´t see that there were two sections here. Should be fixed now, many thanks for spotting!

Zachwang007 commented:
i still can not sovle this problem, my 10.10 jellyfin  tmdb is not working,cannot download metadata

benedikt257 commented:
> i still can not sovle this problem, my 10.10 jellyfin tmdb is not working,cannot download metadata

Hi @Zachwang007, do you have a specific TMDB id for which it is not working?

just to be safe: the changes here are not yet working, unless you built this from source and are running this branch on your Jellyfin server I would not expect the issue to be fixed. It’s not clear from your message.

@Bond-009 I see the stable back port check is failing due to the 10.10.z branch already existing, if I read it right. Is there anything I can do here to resolve this? Thanks!

Zachwang007 commented:
> > i still can not sovle this problem, my 10.10 jellyfin tmdb is not working,cannot download metadata
> 
> Hi @Zachwang007, do you have a specific TMDB id for which it is not working?
> 
> just to be safe: the changes here are not yet merged and released, unless you built this from source and are running this branch on your Jellyfin server I would not expect the issue to be fixed. It’s not clear from your message.
> 
> @Bond-009 I see the stable back port check is failing due to the 10.10.z branch already existing, if I read it right. Is there anything I can do here to resolve this? Thanks!

i got it, I just wait for the next release .
